pf/internal/defaults/defaults.go

```diff
@@ -273,7 +264,7 @@ func (du *defaultsTransform) extendPropertyMapWithDefaults(
 		// using default value for empty property
 		pv, gotDefault := getDefaultValue(ctx,
 			pk,
-			du.resourceByPath(path),
+			du.resourceInstance,
```

Comment:
Calling out this is a small non-refactor but actual change happening here, From/ComputedDefaults will now apply in nested property case (see commit msg).